src/common/config.rs

```diff
@@ -399,35 +294,26 @@ impl TryFrom<&ComponentConfig> for DynamicComponentConfig {
 }
 
 #[derive(Debug)]
-pub enum ConfigType {
-    Static(&'static StaticComponentConfig),
-    Dynamic(DynamicComponentConfig),
+pub enum ConfigType<'a> {
```

Comment:
should we just remove the enum, use `DynamicComponentConfig` directly and rename to `ComponentConfig`?

Reply:
understood, I'm comfortable with that